src/main/java/org/folio/des/service/bursarlegacy/BursarMigrationService.java

```diff
@@ -0,0 +1,448 @@
+package org.folio.des.service.bursarlegacy;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.folio.des.domain.dto.BursarExportDataToken;
+import org.folio.des.domain.dto.BursarExportFilter;
+import org.folio.des.domain.dto.BursarExportFilterAge;
+import org.folio.des.domain.dto.BursarExportFilterCondition;
+import org.folio.des.domain.dto.BursarExportFilterFeeFineOwner;
+import org.folio.des.domain.dto.BursarExportFilterFeeType;
+import org.folio.des.domain.dto.BursarExportFilterPatronGroup;
+import org.folio.des.domain.dto.BursarExportHeaderFooter;
+import org.folio.des.domain.dto.BursarExportJob;
+import org.folio.des.domain.dto.BursarExportTokenConditional;
+import org.folio.des.domain.dto.BursarExportTokenConditionalConditionsInner;
+import org.folio.des.domain.dto.BursarExportTokenConstant;
+import org.folio.des.domain.dto.BursarExportTokenDateType;
+import org.folio.des.domain.dto.BursarExportTokenFeeAmount;
+import org.folio.des.domain.dto.BursarExportTokenFeeDate;
+import org.folio.des.domain.dto.BursarExportTokenFeeMetadata;
+import org.folio.des.domain.dto.BursarExportTokenLengthControl;
+import org.folio.des.domain.dto.BursarExportTokenUserDataOptional;
+import org.folio.des.domain.dto.BursarExportTransferCriteria;
+import org.folio.des.domain.dto.BursarExportTransferCriteriaConditionsInner;
+import org.folio.des.domain.dto.BursarExportTransferCriteriaElse;
+import org.folio.des.domain.dto.ExportTypeSpecificParameters;
+import org.folio.des.domain.dto.Job;
+import org.folio.des.domain.dto.JobWithLegacyBursarParameters;
+import org.folio.des.domain.dto.JobWithLegacyBursarParametersCollection;
+import org.folio.des.domain.dto.LegacyBursarFeeFines;
+import org.folio.des.domain.dto.LegacyBursarFeeFinesTypeMapping;
+import org.folio.des.domain.dto.LegacyBursarFeeFinesTypeMappings;
+import org.folio.des.service.JobService;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class BursarMigrationService {
+  private static final Integer DEFAULT_LIMIT = 10000;
+
+  private static final String BURSAR_EXPORT_MIGRATION_HEADER_CONSTANT = "LIB02";
+
+  public boolean isLegacyJob(JobWithLegacyBursarParameters job) {
+    LegacyBursarFeeFines bursarFeeFines = job
+      .getExportTypeSpecificParameters()
+      .getBursarFeeFines();
+
+    return bursarFeeFines.getDaysOutstanding() != null;
+  }
+
+  public void recreateLegacyJobs(
+    BursarExportLegacyJobService bursarExportLegacyJobService,
+    JobService jobService
+  ) {
+    log.info("searching for legacy jobs");
+    JobWithLegacyBursarParametersCollection response = bursarExportLegacyJobService.get(
+      0,
+      DEFAULT_LIMIT,
+      "status==SCHEDULED"
```

Comment:
it looks a bit incorrect, job created at the moment when quartz reached scheduled time - you basically need to migrate configs (if required), but not scheduled jobs

one more questions, will previously executed jobs (completed or failed statuses) open correctly on UI?

Reply:
All are now converted (jobs, old and new, as well as configs)